db/versions/0017.yml

```diff
@@ -0,0 +1,287 @@
+            'eventType', event_type,
+            'eventId', event_id) as value,
+          1 as version,
+          github_builds.etag as etag
+      from github_builds
+      where
+        github_builds.task_group_id = decode_string_key(taskcluster_github_builds_entities_load.partition_key);
+      end
+  taskcluster_github_builds_entities_create:
+    description: See taskcluster-lib-entities
+    serviceName: github
+    mode: write
+    args: pk text, rk text, properties jsonb, overwrite boolean, version integer
+    returns: uuid
+    body: |-
+      declare
+        new_row github_builds%ROWTYPE;
+      begin
+        select
+          (properties ->> 'organization')::text as organization,
+          (properties ->> 'repository')::text as repository,
+          (properties ->> 'sha')::text as sha,
+          (properties ->> 'taskGroupId')::text as task_group_id,
+          (properties ->> 'state')::text as state,
+          (properties ->> 'created')::timestamptz as created,
+          (properties ->> 'updated')::timestamptz as updated,
+          (properties ->> 'installationId')::integer as installation_id,
+          (properties ->> 'eventType')::text as event_type,
+          (properties ->> 'eventId')::text as event_id,
+          public.gen_random_uuid() as etag
+        into new_row;
+        if overwrite then
+          raise exception 'overwrite not implemented';
+        else
+          execute 'insert into github_builds select $1.*' using new_row;
+        end if;
+        return new_row.etag;
+      end
+  taskcluster_github_builds_entities_scan:
+    description: See taskcluster-lib-entities
+    mode: read
+    serviceName: github
+    args: pk text, rk text, condition text, size integer, page integer
+    returns: table (partition_key text, row_key text, value jsonb, version integer, etag uuid)
+    body: |-
+      declare
+      begin
+        if not condition is null then
+          raise exception 'condition not supported';
+        end if;
+        return query
+          select
+            taskcluster_github_builds_entities_scan.partition_key,
+            'taskGroupId' as row_key,
+            jsonb_build_object(
+              'PartitionKey', encode_string_key(task_group_id),
+              'RowKey', 'taskGroupId',
+              'organization', organization,
+              'repository', repository,
+              'sha', sha,
+              'taskGroupId', task_group_id,
+              'state', state,
+              'created', created,
+              'updated', updated,
+              'installationId', installation_id,
+              'eventType', event_type,
+              'eventId', event_id) as value,
+              1 as version,
+          github_builds.etag as etag
+          from github_builds
+          where
+            partition_key is null or
+            task_group_id = decode_string_key(partition_key)
+          order by task_group_id
+          limit size + 1
+          offset page;
```

Comment:
If size or offset are not defined then something may go wrong here. One way to fix that is to do:

```
        limit case
          when (size is not null and size > 0) then size + 1
          else null
        end
        offset case
          when (page is not null and page > 0) then page
          else 0
        end;
```